common/persistence/visibility/visibility_manager_rate_limited.go

```diff
@@ -228,8 +230,25 @@ func (m *visibilityManagerRateLimited) GetWorkflowExecution(
 	ctx context.Context,
 	request *manager.GetWorkflowExecutionRequest,
 ) (*manager.GetWorkflowExecutionResponse, error) {
-	if ok := m.readRateLimiter.Allow(); !ok {
+	if ok := allow(ctx, "GetWorkflowExecution", CallerSegmentMissing, m.readRateLimiter); !ok {
 		return nil, persistence.ErrPersistenceLimitExceeded
 	}
 	return m.delegate.GetWorkflowExecution(ctx, request)
 }
+
+func allow(
+	ctx context.Context,
+	api string,
+	shardID int32,
```

Comment:
I think this is confusing. Visibility store is not partitioned based on history shardID. I would just leave this field as empty.